### ISSUE TITLE
Service events not received in `Receiver`

### ISSUE THREAD

**Issue by tom-binary** [binary-com/perl-Myriad#119]
With this, I would expect output indicating that events were delivered to the `Receiver`.

However, only the `Emitter` appears to be generating anything:

```perl
package Service::Spam;

use Myriad::Service;

=head2 relentless

A furious storm of neverending events.

Aside from pausing while blocked, and deferring after each message to the next
event loop iteration, this will continuously emit sequentially-ascending
C<< { id => 123 } >> events.

=cut

async method relentless : Emitter() ($sink) {
    my $id = 1;
    while(1) {
        $log->debugf('Wait for later');
        await $self->loop->later;
        $log->debugf('It is now later');
        my $blocked = $sink->unblocked;
        unless($blocked->is_ready) {
            $log->debugf('We were blocked, and will wait');
            await $blocked;
            $log->debugf('We are no longer blocked after %.2fms, and will resume', 1000.0 * $blocked->elapsed);
        }
        $log->debugf('Next ID will be %d', $id);
        $sink->emit({ id => $id++ });
        $log->infof('Emitted and incremented to %d', $id);
        await $self->loop->delay_future(after => 0.5);
    }
}

=head2 burn

The counterpart to L</relentless>, this will receive those events to clear up
space in the queue.

=cut

async method burn : Receiver(channel => 'relentless') ($src) {
    $src->each(sub {
        $log->infof('Have item %s', $_)
    })->retain;
    return $src;
}

1;
```

at `info` level, the output is:

```
Starting Myriad on ... pid ... at 2021-04-27T11:41:50.349351+08:00
Starting service [service.spam]
Done
Emitted and incremented to 2
Have item {data => {id => 1}}
Emitted and incremented to 3
Have item {data => {id => 2}}
Emitted and incremented to 4
Have item {data => {id => 3}}
Emitted and incremented to 5
Have item {data => {id => 4}}
Emitted and incremented to 6
Have item {data => {id => 5}}
```

**Comment by eyadof** [comment]
I'm quite struggling to understand this issue you do have `Have item ....` in the output isn't this from the `Receiver`?

**Comment by tom-binary** [comment]
So the issue here is in the memory transport - the expected output is received when using Redis, but we were only receiving the first item with `--transport=memory`.

This should now be fixed in #124, so I think this can be closed.
